### ISSUE TITLE
Fail to compile Vue Storefront in docker

### ISSUE THREAD

**Issue by cartooncatfish** [DivanteLtd/vue-storefront#4969]
## Current behavior

I followed the document https://docs.vuestorefront.io/guide/cookbook/setup.html#_2-recipe to run the vue storefront on local computer.

At step 11, after running the 
```
docker-compose up -d
```
The http://localhost:3000 has no return.

The docker log shows the following compile errors:

```
... ...
app_1  | Entrypoint app = manifest.js manifest.js.map vendor.js vendor.js.map app.js app.js.map
app_1  | [0] multi webpack-hot-middleware/client @babel/polyfill ./core/client-entry.ts 52 bytes {app} [built]
app_1  | [./core/app.ts] 8.36 KiB {app} [built]
app_1  | [./core/build/config.json] 15.9 KiB {app} [built]
app_1  | [./core/client-entry.ts] 9.62 KiB {app} [built]
app_1  | [./core/hooks.ts] 1.94 KiB {app} [built]
app_1  | [./core/i18n/index.ts] 6.36 KiB {app} [built]
app_1  | [./core/lib/async-data-loader.ts] 1.21 KiB {app} [built]
app_1  | [./core/lib/logger.ts] 9.35 KiB {app} [built]
app_1  | [./core/lib/multistore.ts] 12.4 KiB {app} [built] [1 error]
app_1  | [./core/lib/router-manager.ts] 5.65 KiB {app} [built]
app_1  | [./core/lib/storeCodeFromRoute.ts] 2.21 KiB {app} [built]
app_1  | [./core/lib/sync/task.ts] 15.9 KiB {app} [built]
app_1  | [./core/modules/offline-order/helpers/onNetworkStatusChange.ts] 1.36 KiB {app} [built]
app_1  | [./node_modules/@babel/polyfill/lib/index.js] 686 bytes {vendor} [built]
app_1  | [./node_modules/webpack-hot-middleware/client.js] (webpack)-hot-middleware/client.js 7.68 KiB {vendor} [built]
app_1  |     + 2228 hidden modules
app_1  | 
app_1  | ERROR in /var/www/core/lib/multistore.ts
app_1  | ./core/lib/multistore.ts
app_1  | [tsl] ERROR in /var/www/core/lib/multistore.ts(153,43)
app_1  |       TS2345: Argument of type '{ [key: string]: unknown; }' is not assignable to parameter of type 'Record<string, string | number | boolean | (string | number | boolean)[]>'.
app_1  |   Index signatures are incompatible.
app_1  |     Type 'unknown' is not assignable to type 'string | number | boolean | (string | number | boolean)[]'.
app_1  |       Type '{}' is missing the following properties from type '(string | number | boolean)[]': length, pop, push, concat, and 26 more.
app_1  | 
app_1  | ERROR in /var/www/core/lib/search/adapter/api/searchAdapter.ts
app_1  | ./core/lib/search/adapter/api/searchAdapter.ts
app_1  | [tsl] ERROR in /var/www/core/lib/search/adapter/api/searchAdapter.ts(76,45)
app_1  |       TS2345: Argument of type 'HttpQuery' is not assignable to parameter of type 'Record<string, string | number | boolean | (string | number | boolean)[]>'.
app_1  |   Index signature is missing in type 'HttpQuery'.
app_1  | 
app_1  | ERROR in /var/www/core/lib/search/adapter/api-search-query/searchAdapter.ts
app_1  | ./core/lib/search/adapter/api-search-query/searchAdapter.ts
app_1  | [tsl] ERROR in /var/www/core/lib/search/adapter/api-search-query/searchAdapter.ts(84,45)
app_1  |       TS2345: Argument of type 'HttpQuery' is not assignable to parameter of type 'Record<string, string | number | boolean | (string | number | boolean)[]>'.
app_1  | Child html-webpack-plugin for "index.amp.html":
app_1  |              Asset     Size  Chunks  Chunk Names
app_1  |     index.amp.html  535 KiB       0  
app_1  |     Entrypoint undefined = index.amp.html
app_1  |     [./node_modules/html-webpack-plugin/lib/loader.js!./src/index.amp.template.html] 660 bytes {0} [built]
app_1  |     [./node_modules/lodash/lodash.js] 528 KiB {0} [built]
app_1  |     [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {0} [built]
app_1  |     [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
app_1  | Child html-webpack-plugin for "index.basic.html":
app_1  |                Asset     Size  Chunks  Chunk Names
app_1  |     index.basic.html  535 KiB       0  
app_1  |     Entrypoint undefined = index.basic.html
app_1  |     [./node_modules/html-webpack-plugin/lib/loader.js!./src/themes/default/templates/index.basic.template.html] 723 bytes {0} [built]
app_1  |     [./node_modules/lodash/lodash.js] 528 KiB {0} [built]
app_1  |     [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {0} [built]
app_1  |     [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
app_1  | Child html-webpack-plugin for "index.html":
app_1  |          Asset     Size  Chunks  Chunk Names
app_1  |     index.html  535 KiB       0  
app_1  |     Entrypoint undefined = index.html
app_1  |     [./node_modules/html-webpack-plugin/lib/loader.js!./src/themes/default/templates/index.template.html] 853 bytes {0} [built]
app_1  |     [./node_modules/lodash/lodash.js] 528 KiB {0} [built]
app_1  |     [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {0} [built]
app_1  |     [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
app_1  | Child html-webpack-plugin for "index.minimal.html":
app_1  |                  Asset     Size  Chunks  Chunk Names
app_1  |     index.minimal.html  535 KiB       0  
app_1  |     Entrypoint undefined = index.minimal.html
app_1  |     [./node_modules/html-webpack-plugin/lib/loader.js!./src/themes/default/templates/index.minimal.template.html] 697 bytes {0} [built]
app_1  |     [./node_modules/lodash/lodash.js] 528 KiB {0} [built]
app_1  |     [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {0} [built]
app_1  |     [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
app_1  | ℹ ｢wdm｣: Failed to compile.
```

How can I solve it? It seems to relate to the default settings.

Thanks

## Steps to reproduce the issue

I followed the steps in the document except in step 7 I chose the latest one, the version 1.12.2.

## Version of Vue Storefront

- [X] Vue Storefront
In the package.json, the version information is:
  "name": "vue-storefront",
  "version": "1.12.2",
- [ ] Vue Storefront Next

## Can you handle fixing this bug by yourself?

- [ ] YES
- [X] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer. (doesn't apply to Next)
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: 
   Chrome/Firefox
- OS: 
   Mac version 10.15.6
- Node: 
- Code Version: 
- Docker Version: 2.3.0.4

**Comment by septoctobre** [comment]
@cartooncatfish  You can look in #4951

**Comment by Fifciu** [comment]
Recently, @AdamPawlinski had this problem and he says reinstalling project solved it.

In my opinion, in case of docker you might want to add:
```
- './yarn.lock:/var/www/yarn.lock'
```
To the docker-compose's volumes part.

**Comment by gibkigonzo** [comment]
solved by #4951
